genshin/client/components/hoyolab.py

```diff
@@ -29,6 +29,25 @@ async def search_users(
         )
         return [models.PartialHoyolabUser(**i["user"]) for i in data["list"]]
 
+    async def get_hoyolab_self(self, *, lang: typing.Optional[str] = None) -> models.PartialHoyolabUser:
+        """Get hoyolab data for me."""
+        if self.region == types.Region.OVERSEAS:
+            url = "https://bbs-api-os.hoyolab.com/community/painter/wapi/user/full"
+            referer = "https://www.hoyolab.com/"
+        elif self.region == types.Region.CHINESE:
+            url = "https://bbs-api.mihoyo.com/user/wapi/getUserFullInfo?gids=2"
```

Comment:
Please use `routes.py` instead of putting full urls in methods.

Reply:
And request_hoyolab uses routes.TAKUMI_URL, which doesn't work because their endpoint is ```https://bbs-api-os.hoyolab.com``` and ```https://bbs-api.mihoyo.com```, not 
```python
TAKUMI_URL = InternationalRoute(
    overseas="https://api-os-takumi.mihoyo.com/",
    chinese="https://api-takumi.mihoyo.com/",
)
```

Reply:
Yeah you can make a new route definition. I use a full urls because it's old code I haven't gotten around to clean yet.

Reply:
Yes, if requested using the request_hoyolab function, the server will return `invalid request`. We also need another function for request. It'll need to use [DS](https://github.com/thesadru/genshin.py/blob/8c71c0d9652137a40a96dcf6922176ee573bfa38/genshin/utility/ds.py#L17) algorithm as opposed to [DS2](https://github.com/thesadru/genshin.py/blob/8c71c0d9652137a40a96dcf6922176ee573bfa38/genshin/utility/ds.py#L25).

Reply:
Yeah why not. They could both use a function like `request_ds`.

Reply:
Done.